server/src/main/java/org/elasticsearch/common/blobstore/BlobPath.java

```diff
@@ -62,6 +63,12 @@ public BlobPath add(String path) {
         return new BlobPath(Collections.unmodifiableList(paths));
     }
 
+    public BlobPath append(BlobPath otherPath) {
```

Comment:
Generally I would prefer we keep utilities like these as methods rather than functions in other places. This way we avoid duplicating code.

But I understand your dislike of creeping small utility functions, so I've removed all the utilities and instead relied on the iterator.